### ISSUE TITLE
About training speed

### ISSUE THREAD

**Issue by ghy0324** [nihalsid/panoptic-lifting#18]
Hi! Thanks for your excellent work!
As stated in your paper, you train on A6000 for 450k iterations (about 10 hours). I am currently running the default configuration on the `itw` dataset you provide and found that there are 10 epochs in total, each with 130k iterations (requiring more than 3 hours for each epoch). Is this due to the large size of this dataset?
I want to get results faster, allowing for a slight sacrifice in accuracy, so I made the following modifications: 1. set `max_epoch` to `3`; 2. set `instance_optimization_epoch` and `segment_optimization_epoch` to `1` and `2`.
Is this reasonable? And do I need to make any other modifications?

**Comment by xingy038** [comment]
I use a single 3090, it will spend more than 7 hours when after 6 epoch. And the multi-GPU is training when after 3 epoch will be stuck.

**Comment by nihalsid** [comment]
Hi!

Thanks for your interest in the work and the issue. Unfortunately, I encountered the same issue which was introduced at some point during multi-gpu training, which stems from syncing across gpus after "on validation end" step. I didn't invest time in debugging this and we trained on a single gpu only. 

The 10 hour time is for hypersim scenes and some scannet scenes. For scenes with a large number of frames, the time can go quite large. One work around could be using a lower resolution (384) but that will harm the quality slightly.

Setting max_epoch to 3 might actually make the quality quite lower, perhaps try the lower resolution option first, maybe that's enough?

Thanks
